crates/polars-ops/src/series/ops/replace.rs

```diff
@@ -89,8 +102,12 @@ fn replace_by_multiple(
             replaced.zip_with(mask, default)
         },
         Err(_) => {
-            let mask = &replaced.is_not_null();
-            replaced.zip_with(mask, default)
+            if replaced.null_count() > 0 {
```

Comment:
Ai.. good fast path.

Reply:
Yes, but it can be moved outside of the match block! We can do the check directly after line 97 and early return.